packages/react-router-dom/index.tsx

```diff
@@ -773,10 +773,7 @@ function useSubmitImpl(fetcherKey?: string, routeId?: string): SubmitFunction {
 
       let href = url.pathname + url.search;
       let opts = {
-        // If replace is not specified, we'll default to false for GET and
-        // true otherwise
-        replace:
-          options.replace != null ? options.replace === true : method !== "get",
+        replace: options.replace,
```

Comment:
This probably shouldn't have been at this level in the first place, since that makes it react-specific, even though it's not related to the UI at all